atp/client.go

```diff
@@ -259,8 +295,19 @@ func (c *client) executeWriteLoop(
 	runID string,
 	receivedSignals chan schema.Input,
 ) {
-	// Add the channel to the client so that it can be kept track of
 	c.mutex.Lock()
+	if c.done {
+		c.mutex.Unlock()
+		// Close() was called, so exit now to prevent the channel from being added to
+		// the map, since Close() uses that map to determine the exit condition.
+		// Adding to the map would cause it to never exit.
+		c.logger.Warningf(
```

Comment:
Is this really a "warning"? This runs in the background during a step; is it true that go func should always have been shut down by the step before `c.done` can set, or is this just an alternate but "normal" termination path for the background thread?